File: problems/problems_163/solution.ts
function findMissingRanges(nums: number[], lower: number, upper: number): number[][] {
	const ans: number[][] = [];
	nums.push(upper + 1);
	let last: number = lower - 1;
	for (const num of nums) {
		if (num - last > 1) {
			ans.push([last + 1, num - 1]);
		}
		last = num;
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const splits: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(splits[0]);
	const lower: number = JSON.parse(splits[1]);
	const upper: number = JSON.parse(splits[2]);
	return findMissingRanges(nums, lower, upper);
}

File: problems/problems_163/solution.go
package problem163

import (
	"encoding/json"
	"log"
	"strings"
)

func findMissingRanges(nums []int, lower int, upper int) (ans [][]int) {
	nums = append(nums, upper+1)
	last := lower - 1
	for _, num := range nums {
		if d := num - last; d > 1 {
			ans = append(ans, []int{last + 1, num - 1})
		}
		last = num
	}
	return
}

func Solve(input string) any {
	values := strings.Split(input, "\n")
	var nums []int
	var lower int
	var upper int

	if err := json.Unmarshal([]byte(values[0]), &nums); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(values[1]), &lower); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(values[2]), &upper); err != nil {
		log.Fatal(err)
	}

	return findMissingRanges(nums, lower, upper)
}
